bind.go

```diff
@@ -103,7 +103,22 @@ func (b *DefaultBinder) bindData(ptr interface{}, data map[string][]string, tag
 				continue
 			}
 		}
-		inputValue, exists := data[inputFieldName]
+
+		var inputValue []string
+		var exists bool
+
+		// Go json.Unmarshal supports case insensitive binding.  However the url
+		// params are bound case sensitive which is inconsistent.  To fix this
+		// we must check all of the map values in a case-insensitive search.
```

Comment:
The [package docs for `json`](https://golang.org/pkg/encoding/json/) has:

> Unmarshal matches incoming object keys to the keys used by Marshal (either the struct field name or its tag), preferring an exact match but also accepting a case-insensitive match.

So I think that to match the `json.Unmarshal` behaviour, it should be something like:

```go
inputValue, exists := data[inputFieldName]
if !exists {
	inputFieldName = strings.ToLower(inputFieldName)
	for k, v := range data {
		if strings.ToLower(k) == inputFieldName {
			inputValue = v
			exists = true
			break
		}
	}
}
if !exists {
	continue
}
```